### PR TITLE
11783 concrete design addon

### PR DESCRIPTION
**Pull request by david-liska**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [ ] Attached examples
- [ ] Own example attached to Pull Request

**Test Configuration**:
* RFEM / RSTAB / RSECTION version:
# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: includes/AddOns/TypesForConcreteDesign/SurfaceReinforcement.js [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+    if (typeof additional_rebar_spacing_auto_priority !== "undefined") {
+        this.surface_reinforcement.additional_rebar_spacing_auto_priority = additional_rebar_spacing_auto_priority;
+    }
+};
+
+/**
+ * Sets Stirrups diameter
+ * @param {Number} stirrup_diameter     Stirrups diameter
+ */
+ConcreteDesignSurfaceReinforcement.prototype.SetStirrupsDiameter = function (value_1) {
```
Comment (jarabroz) [inline includes/AddOns/TypesForConcreteDesign/SurfaceReinforcement.js:408]:
```suggestion
ConcreteDesignSurfaceReinforcement.prototype.SetStirrupsDiameter = function (diameter) {
```

File: includes/AddOns/TypesForConcreteDesign/SurfaceReinforcement.js [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+    ASSERT(this.surface_reinforcement.reinforcement_type === surface_reinforcements.REINFORCEMENT_TYPE_STIRRUPS, "Reinforcement must be of stirrups type");
+    //this.surface_reinforcement.stirrup_diameter_auto_enabled = false;   // Can't set (bug 94787 - comment)?
+    if (IsCurrentCodeOfStandard("EN") || IsCurrentCodeOfStandard("SP")) {
+        ASSERT(typeof value_1 !== "undefined", "Stirrups diameter must be specified");
+        this.surface_reinforcement.stirrup_diameter = value_1;
```
Comment (jarabroz) [inline includes/AddOns/TypesForConcreteDesign/SurfaceReinforcement.js:413]:
```suggestion
        this.surface_reinforcement.stirrup_diameter = diameter;
```

**Review by jarabroz — CHANGES_REQUESTED** [review]

File: includes/AddOns/ConcreteDesign/ConcreteDesignSupport.js [hunk 3]
```diff
@@ -0,0 +1,1114 @@
+include("../../Tools/jshlf_common_functions.js");
```
Comment (jarabroz) [inline includes/AddOns/ConcreteDesign/ConcreteDesignSupport.js:1]: Please consider renamig of the file - misleading name - it contains auxiliary functions but looks like something else.

File: includes/AddOns/ConcreteDesign/ServiceabilityConfigurationNTC.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+/*
+NO API support?
```
Comment (jarabroz) [inline includes/AddOns/ConcreteDesign/ServiceabilityConfigurationNTC.js:2]: Why?

File: includes/AddOns/ConcreteDesign/UltimateConfigurationEN.js [hunk 5]
```diff
@@ -0,0 +1,1330 @@
...
+            break;
+        case "WALLS":
+            addon.settings_surface_ec2.property_maximum_longitudinal_reinforcement_for_walls = true;
+            break;
+        default:
+            ASSERT(false, "SetConcreteDesignReinforcementMaximumLongitudinalReinforcementType: unknown maximum longitudinal reinforcement type");
+    }
+}
+
+//Grandmaster version newly not supports this
```
Comment (jarabroz) [inline includes/AddOns/ConcreteDesign/UltimateConfigurationEN.js:1156]: Do we need it? Can we delete it?

**Review by jarabroz — CHANGES_REQUESTED** [review]

File: examples/high_level_functions_demos/addon_concrete_design.js [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+
+if (RFEM) {
+    /*************************************************** Types for concrete design - Reinforcement direction ****************************************************/
+    var reinforcementDirection = new ConcreteDesignReinforcementDirection(undefined, [surfaceList[4].no, surfaceList[5].no], "Concrete design Reinforcement direction");
+    reinforcementDirection.SetName("Reinforcement direction for test");
+    reinforcementDirection.SetDirectionType("ROTATED");
+    reinforcementDirection.SetDirectionRotations(Math.PI / 4, Math.PI / 2);
+
+    /**************************************************** Types for concrete design - Surface reinforcement ****************************************************/
+    var reinforcementMaterial = new Material(undefined, "Grade 40");
```
[the remark this answers is not in the document]
Comment (david-liska) [reply examples/high_level_functions_demos/addon_concrete_design.js]: It seems you fixed it?

File: examples/high_level_functions_demos/addon_concrete_design.js [hunk 7]
```diff
@@ -0,0 +1,511 @@
+include("../../includes/Tools/high_level_functions_support.js");
+include("../../includes/AddOns/ConcreteDesign/ConcreteDesignSupport.js");
+/*********************************************************************************************
+****************************************** Main **********************************************
+*********************************************************************************************/
+run("../includes/Tools/clearAll.js");
+
+var material = new Material(undefined, "C12/15");
+var section = new Section(undefined, "IPE 80", material.GetNo());
```
[the remark this answers is not in the document]
Comment (david-liska) [reply examples/high_level_functions_demos/addon_concrete_design.js]: It seems you fixed it?

File: examples/high_level_functions_demos/addon_concrete_design.js
[the remark this answers is not in the document]
Comment (david-liska) [reply examples/high_level_functions_demos/addon_concrete_design.js:1]: Steel design is based on this idea too and finished AT is design with this way. We can discuss about it.

File: includes/AddOns/ConcreteDesign/ConcreteDesignSupport.js (hunk 3 above)
Comment (david-liska) [reply includes/AddOns/ConcreteDesign/ConcreteDesignSupport.js:1]: File removed, function EnumValueFromJSHLFTypeName moved to high_level_functions_support.js.

File: includes/AddOns/ConcreteDesign/UltimateConfigurationNTC.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+/*
```
[the remark this answers is not in the document]
Comment (david-liska) [reply includes/AddOns/ConcreteDesign/UltimateConfigurationNTC.js:1]: File removed.

File: includes/AddOns/ConcreteDesign/UltimateConfigurationEN.js (hunk 5 above)
Comment (david-liska) [reply includes/AddOns/ConcreteDesign/UltimateConfigurationEN.js:1156]: Deleted.

**Review by jarabroz — APPROVED** [review]

**Review by jarabroz — APPROVED** [review]

**Pull request merged**